src/backend/cdb/cdbtm.c

```
@@ -1515,6 +1515,7 @@ initTM(void)
 					 "looking for failed segments.");
 
 				/* Log the error. */
+				elog_demote(LOG);
```

Suppressing an error like this is very dangerous in general. If an error happens, all the backend code assumes that an abort will happen shortly. If you don't abort the transaction, you might leak locks and other resources. In the worst case, you might have e.g. inserted a tuple but not its index entries yet. This code probably doesn't make on-disk changes like that, but still.

So at the very least this needs a long comment explaining why this TRY-CATCH block is needed, and why it's OK to do what it does. Or more likely, this ought to be rewritten in a more safe way, perhaps by opening & closing a new transaction around this.

What process runs initTM()? When does it run, in the middle of another transaction, or in a background process that does nothing else than probe QEs? Or in crash recovery?

---

@hlinnaka This code change is not suppressing the ERROR, that was done earlier as well. The change is more extension of discussion we had in PR https://github.com/greenplum-db/gpdb/pull/3289.

InitTM is run as part of first backend created after DB start. And in case `ERROR` is hit on first attempt but succeeds on next attempt the command/query completes fine but still message with ERROR is written to log, plus sent to client. This becomes point of confusion as only reason we are logging the message is for debugging or troubleshooting purpose and since on retry it succeeded it should not get reported to client. Hence just writing the message with LOG avoids that confusion instead of retaining the original LOG level.

---

The first backend process that wants to execute a distributed transaction runs runs `initTM()`.  @ashwinstar not in this PR but I think the retry logic should be removed from `initTM()`.  Let the transaction abort and let the user run another query which will result in another attempt at `initTM()`.